lib/errors/index.js

```diff
@@ -11,7 +11,6 @@ class BaseError extends Error {
   constructor(message) {
     super(message);
     this.name = 'SequelizeBaseError';
-    this.message = message;
```

Comment:
The super class should already set `this.message = message`. The only slight API change is if clients relied on Sequelize's behavior of **not** defaulting `message` to '', which Error does.

Reply:
ECMAScript specs says error messages should default to empty string where we are setting them to `undefined`.

Bluebird's error check also conforms to this, but it would be better if they also include `instance of` check as undefined error message very much is possible in everyday code.